app/services/badge_granter.rb

```diff
@@ -446,6 +446,8 @@ def self.backfill(badge, opts = nil)
 
       notification = send_notification(row.user_id, row.username, row.locale, badge)
 
+      DiscourseEvent.trigger(:user_badge_granted, badge.id, row.user_id)
```

Comment:
Instead of duplicating this here and in the `UserBadge` model, I think we should consolidate the logic here into a method that can be called by this server and the model. Otherwise, this makes the event brittle to future changes when we want to add more arguments to the trigger.

Reply:
> Instead of duplicating this here and in the `UserBadge` model, I think we should consolidate the logic here into a method that can be called by this server and the model. Otherwise, this makes the event brittle to future changes when we want to add more arguments to the trigger.

What do you have in mind?

`badge.granted_to_user_id(user_id)`

```
# badge.rb
def granted_to_user_id(user_id)
  DiscourseEvent.trigger(:user_badge_granted, self.id, user_id)
end
```

?

Reply:
Yup or even just `trigger_badge_granted_event`